account-kit/core/src/store/types.ts

```diff
@@ -81,15 +82,23 @@ export type StoreState = {
   signer?: AlchemyWebSigner;
   accounts?: {
     [chain: number]: {
-      [key in SupportedAccountTypes]: AccountState<key>;
+      [T in SupportedAccountTypes]: T extends "ModularAccountV2"
```

Comment:
i am slightly debating if it might be worth just nesting the accounts, configs, and clients for each `type` under a `mode`, even if they don't have different modes. it would just be `default` for all types other than MAv2. if people are okay w/ that, it would lead to less branching in a lot of places where we have to check the `type` to determine how we need to access the final objects.

Reply:
I think this is not only an easier (for future devs) but probably safer option.